packages/core-ethereum/src/channel/index.ts

```diff
@@ -50,28 +49,42 @@ class ChannelFactory {
     const self = new Public(this.coreConnector.account.keys.onChain.pubKey)
     const selfAddress = await self.toAddress()
 
-    indexer.on('channelOpened', async ({ partyA: _partyA, partyB: _partyB, channelEntry }: ChannelUpdate) => {
-      const partyA = new Public(_partyA)
-      const partyAAddress = await partyA.toAddress()
-      const partyB = new Public(_partyB)
+    indexer.on('channelOpened', async (channel: ChannelEntry) => {
+      const accountAPubKey = await this.coreConnector.indexer.getPublicKeyOf(channel.parties[0])
+      const accountBPubKey = await this.coreConnector.indexer.getPublicKeyOf(channel.parties[1])
+      if (!accountAPubKey || !accountBPubKey) {
+        log(chalk.red('Currently opening a channel with an unintialized account is not supported'))
```

Comment:
we need to use a single source of truth to track channel state, then we can query that truth and find out which channels can be used to redeem tickets - for now we can force initialization requirement within the SC but seeing we are refactoring things I haven't made that change yet

Reply:
Does that also mean that if events are reordered that we just drop a channel opening / closing?

Reply:
@robertkiel we only process old events which pass the possible reorg limit, we have been doing that since forever, not sure why u don't know this already